proto/cosmos/crypto/sr25519/keys.proto

```diff
@@ -0,0 +1,17 @@
+syntax = "proto3";
+package cosmos.crypto.sr25519;
+
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys/sr25519";
+
+// PubKey defines a sr25519 public key
+message PubKey {
+    option (gogoproto.goproto_stringer) = false;
+    bytes key = 1;
+}
+
+// PrivKey defines a sr25519 private key.
+message PrivKey {
+    bytes miniSecretKey = 1;
```

Comment:
Many key operations (eg: PubKey, Sign, Verify...) require casting (and extra allocation). This could be optimized by using `casttype` option (or `customtype` - whichever will work better). Here is an example: https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/crypto/ed25519/keys.proto#L16

feel free to chat with me - I did it recently for `secp2256r1`

Reply:
This way is fine. In the future when we do migrate off of gogoproto we will be doing this anyways.

Reply:
yes, but why doing all unnecessary u allocations and casting. We already have an example how to do it correctly in `ed25519`.